src/java/org/apache/cassandra/config/DataRateSpec.java

```diff
@@ -0,0 +1,376 @@
+            {
+                return d / 1024;
+            }
+
+            public double toMebibytesPerSecond(double d)
+            {
+                return d / (1024 * 1024);
+            }
+
+            public double toMegabitsPerSecond(double d) { return d / 125000; }
+
+            public double convert(double source, DataRateUnit sourceUnit)
+            {
+                return sourceUnit.toBytesPerSecond(source);
+            }
+        },
+        KIBIBYTES_PER_SECOND("KiB/s")
+        {
+            public double toBytesPerSecond(double d)
+            {
+                return x(d, 1024, MAX / 1024.0);
+            }
+
+            public double toKibibytesPerSecond(double d)
+            {
+                return d;
+            }
+
+            public double toMebibytesPerSecond(double d)
+            {
+                return d / 1024;
+            }
+
+            public double toMegabitsPerSecond(double d)
+            {
+                return d / 122;
+            }
+
+            public double convert(double source, DataRateUnit sourceUnit)
+            {
+                return sourceUnit.toKibibytesPerSecond(source);
+            }
+        },
+        MEBIBYTES_PER_SECOND("MiB/s")
+        {
+            public double toBytesPerSecond(double d)
+            {
+                return x(d, 1024 * 1024, MAX / (1024.0 * 1024.0));
+            }
+
+            public double toKibibytesPerSecond(double d)
+            {
+                return x(d, 1024, MAX / (1024.0));
+            }
+
+            public double toMebibytesPerSecond(double d)
+            {
+                return d;
+            }
+
+            public double toMegabitsPerSecond(double d)
+            {
+                if (d > MAX / (MEGABITS_PER_MEBIBYTE))
+                    return MAX;
+                return Math.round(d * MEGABITS_PER_MEBIBYTE);
+            }
+
+            public double convert(double source, DataRateUnit sourceUnit)
+            {
+                return sourceUnit.toMebibytesPerSecond(source);
+            }
+        };
+
+        static final double MAX = Long.MAX_VALUE;
+        static final double MEGABITS_PER_MEBIBYTE = 8.388608;
+
+        /**
+         * Scale d by m, checking for overflow. This has a short name to make above code more readable.
+         */
+        static double x(double d, double m, double over)
+        {
+            if (d > over)
+                return MAX;
+            return d * m;
+        }
+
+        /**
+         * @param symbol the unit symbol
+         * @return the rate unit corresponding to the given symbol
+         */
+        public static DataRateUnit fromSymbol(String symbol)
+        {
+            for (DataRateUnit value : values())
+            {
+                if (value.symbol.equalsIgnoreCase(symbol))
+                    return value;
+            }
+            throw new IllegalArgumentException(String.format("Unsupported data rate unit: %s. Supported units are: %s",
```

Comment:
@michaelsembwever I will change this exception also to ConfigurationException on commit, I missed it